api/end_device.proto

```diff
@@ -122,7 +122,10 @@ message MACParameters {
     bool enable_uplink = 5;
   }
   // Configured uplink channels and optionally Rx1 frequency.
-  repeated Channel channels = 19 [(validate.rules).repeated.min_items = 1];
+  repeated Channel channels = 19 [(validate.rules).repeated = {
+    min_items: 1,
+    items: { message: { required: true } }
```

Comment:
Wait, is it possible to receive `nil`s in repeated messages received over RPC? Because then we probably need to check this in more places.

Reply:
No, it is not. I'm actually in the wrong here, it is not possible neither via gRPC, nor via the gRPC bridge. The `nil` channel must be from somewhere else.